whois-client/src/main/java/net/ripe/db/whois/common/sso/AuthServiceClient.java

```diff
@@ -0,0 +1,197 @@
+import net.ripe.db.whois.common.sso.domain.ValidateTokenResponse;
+import org.apache.commons.lang.StringUtils;
+import org.glassfish.jersey.client.ClientProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nullable;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.ForbiddenException;
+import javax.ws.rs.InternalServerErrorException;
+import javax.ws.rs.NotAuthorizedException;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.ProcessingException;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
+import static javax.ws.rs.core.Response.Status.UNAUTHORIZED;
+
+@Component
+public class AuthServiceClient {
+
+    public static final String TOKEN_KEY = "crowd.token_key";
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuthServiceClient.class);
+
+    private static final String VALIDATE_PATH = "/authorisation-service/v2/authresource/validate";
+    private static final String ORGANISATION_MEMBERS_PATH = "/authorisation-service/v2/authresource/members";
+    private static final String USER_SEARCH_PATH = "/authorisation-service/v2/authresource/accounts/";
+
+    private static final String CONTACT_PATH = "/contacts";
+    private static final String EMAIL_PATH = "/email";
+    private static final String VALIDATE_TOKEN_PERMISSION = "portal";
+
+    private static final int CLIENT_CONNECT_TIMEOUT = 10_000;
+    private static final int CLIENT_READ_TIMEOUT = 60_000;
+
+    private final Client client;
+
+    private final String restUrl;
+    private final String apiKey;
+
+    @Autowired
+    public AuthServiceClient(
+            @Value("${authorisation.service.api.url:}") final String restUrl,
+            @Value("${authorisation.service.api.key:}") final String apiKey) {
+        this.restUrl = restUrl;
+        this.apiKey = apiKey;
+
+        final JacksonJsonProvider jsonProvider = (new JacksonJaxbJsonProvider())
+                .configure(DeserializationFeature.UNWRAP_ROOT_VALUE, false)
+                .configure(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY, true);
+        this.client = (ClientBuilder.newBuilder()
+                .register(jsonProvider))
+                .property(ClientProperties.CONNECT_TIMEOUT, CLIENT_CONNECT_TIMEOUT)
+                .property(ClientProperties.READ_TIMEOUT, CLIENT_READ_TIMEOUT)
+                .build();
+    }
+
+    @Nullable
+    public ValidateTokenResponse validateToken(final String crowdToken) {
```

Comment:
rename `crowdToken` to `token`, don't mention Crowd.

Reply:
I can rename but the cookie that we get from client still has crowd.token_key, thats why I dint rename this

Reply:
We must leave the cookie name as-is, but I think we should remove 'crowd' where we can, and I think 'token' still makes sense as a method argument.